pkgs/misc/meson64-tools/default.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildPackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "meson64-tools";
+  version = "unstable-2023-07-25";
+
+  src = fetchFromGitHub {
+    owner = "angerman";
+    repo = pname;
+    rev = "b09cefd1e001dbba14036857bf6e167bf1833f26";
+    hash = "sha256-/koIsslDNpaFHf1TV/0Xt0TiyhjL6tCz2oHQraYNhPA=";
+  };
+
+  nativeBuildInputs = with buildPackages; [ openssl bison yacc flex bc python3 ];
+
+  preBuild = ''
+    patchShebangs --build .
+    patchShebangs --build ./mbedtls/scripts/generate_psa_constants.py
+    substituteInPlace mbedtls/programs/fuzz/Makefile --replace "python2" "python"
+    substituteInPlace mbedtls/tests/Makefile --replace "python2" "python"
+  '';
+
+  # Also prefix tool names since some names are really generic (e.g. `pkg`).
+  # Otherwise something could shadow those generic names in other builds.
+  postInstall = ''
+    (cd $out/bin
+      for bin in *; do
+        ln -s $bin meson64-$bin
```

```suggestion
        ln -s $bin meson-g12-$bin
```

Since then, I've learned more about Amlogic (meson). There are families of hardware (different SoCs configured broadly the same) and those families can be grouped in "supersets" for how they work

 - https://linux-meson.com/hardware.html#supported-wip-soc-families

As far as I know, up to today, there are only "super-families", `gx` and `g12`. (I suspect gx is g10, follow-up from meson8, which was a follow-up from meson6?)

This would better represent what the tool does, since it's a replacement for `aml_encrypt_g12b`, which is used basically for all g12 family devices.